src/main/js/main_html5.js

```diff
@@ -1620,6 +1464,271 @@ require("../../../html5-common/js/utils/environment.js");
+
+          if (language === targetLanguage) {
+            wasTargetTrackAdded = true;
+          }
+        }
+      }
+      return wasTargetTrackAdded;
+    };
+
+    /**
+     * Creates a single TextTrack object using the values provided in trackData.
+     * The new track's mode will be set to targetMode after creation if the track
+     * matches targetLanguage. Tracks that don't match targetLanguage will have a
+     * 'disabled' mode by default.
+     * @private
+     * @method OoyalaVideoWrapper#addExternalCaptionsTrack
+     * @param {Object} trackData An object with the following properties:
+     *  - url: {String} The url of a source VTT file
+     *  - name: {String} The label to display for this track
+     *  - language: {String} The language code of the closed captions
+     * @param {String} targetLanguage The language or key of the track that should be set to targetMode
+     * (usually the language that should be active).
+     * @param {String} targetMode The mode that should be set on the track that matches targetLanguage.
+     */
+    const addExternalCaptionsTrack = (trackData = {}, targetLanguage, targetMode) => {
+      let trackMode;
+      // Disable new tracks by default unless their language matches the language
+      // that is meant to be active
+      if (trackData.language === targetLanguage) {
+        trackMode = targetMode;
+      } else {
+        trackMode = OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED;
+      }
+      // Keep a record of all the tracks that we add
+      const trackId = textTrackMap.addEntry({
+        src: trackData.url,
+        label: trackData.name,
+        language: trackData.language,
+        mode: trackMode
+      }, true);
+      // Create the actual TextTrack object
+      textTrackHelper.addTrack({
+        id: trackId,
+        kind: CONSTANTS.TEXT_TRACK.KIND.SUBTITLES,
+        // IMPORTANT:
+        // We initially set the label to trackId since it's the only
+        // cross-browser way to indentify the track after it's created
+        label: trackId,
+        srclang: trackData.language,
+        src: trackData.url
+      });
+    };
+
+    /**
+     * Registers unknown text tracks in our text track map and ensures that
+     * any tracks that we add have the track mode that corresponds to them.
+     * This method is called when there are text track changes such as when the
+     * addtrack or removetrack events are fired.
+     * @private
+     * @method OoyalaVideoWrapper#tryMapTextTracks
+     */
+    const tryMapTextTracks = () => {
+      textTrackHelper.forEach(textTrack => {
+        // Any tracks that have a track id as a label are known to be external
+        // VTT tracks that we recently added. We rely on the label as the only
+        // cross-browser way to identify a TextTrack object after its creation
+        const trackMetadata = textTrackMap.findEntry({
+          id: textTrack.label
+        });
+
+        if (trackMetadata) {
```

Comment:
What happens if the trackMetadata isn't found?

Reply:
This would just mean that either all external tracks have already been mapped or the `addtrack` event was triggered by an in-stream track addition. This call is meant to be idempotent so that it can be executed at any point in time